External/Plugins/CodeRefactor/PluginMain.cs

```diff
@@ -655,7 +661,79 @@ public void SaveSettings()
             ObjectSerializer.Serialize(this.settingFilename, this.settingObject);
         }
 
+        void OnAddRefactorOptions(List<ICompletionListItem> list)
+        {
+            if (list == null)
+                return;
+
+            RefactorItem.AddItemToList(refactorMainMenu.RenameMenuItem, list);
+            RefactorItem.AddItemToList(refactorMainMenu.ExtractMethodMenuItem, list);
+            RefactorItem.AddItemToList(refactorMainMenu.ExtractLocalVariableMenuItem, list);
+            RefactorItem.AddItemToList(refactorMainMenu.DelegateMenuItem, list);
+
+            var features = ASContext.Context.Features;
+
+            if (!features.hasImports)
+                return;
+
+            var sci = ASContext.CurSciControl;
+            string line = sci.GetLine(sci.CurrentLine).TrimStart();
+
+            if (line.StartsWith(features.importKey, StringComparison.Ordinal)
+                || !string.IsNullOrEmpty(features.importKeyAlt) && line.StartsWith(features.importKeyAlt, StringComparison.Ordinal))
+            {
+                RefactorItem.AddItemToList(refactorMainMenu.OrganizeMenuItem, list);
+
+                if (features.hasImportsWildcard)
+                    RefactorItem.AddItemToList(refactorMainMenu.TruncateMenuItem, list);
+            }
+        }
+
         #endregion
 
     }
+
+    class RefactorItem : ICompletionListItem
+    {
+        ToolStripItem item;
+        string label;
+        Bitmap icon;
+
+        public static void AddItemToList(ToolStripMenuItem item, List<ICompletionListItem> list)
+        {
+            if (item.Enabled)
+                list.Add(new RefactorItem(item));
+        }
+
+        public RefactorItem(ToolStripItem item)
+        {
+            this.item = item;
+            label = Regex.Replace(item.Text, "[&.]", string.Empty);
+            icon = new Bitmap(item.Image ?? PluginBase.MainForm.FindImage("452")); //452, 473
+        }
+
+        public string Description
+        {
+            get { return label; /*TextHelper.GetString("ASCompletion.Info.GeneratorTemplate");*/ }
```

Comment:
I don't know which is more appropriate between `label` and `TextHelper.GetString("ASCompletion.Info.GeneratorTemplate")` as the description. I decided that the refactor items would be better to have different descriptions than the original context generation options, so I used `label`.